packages/controller/lib/setup/setupBackup.js

```diff
@@ -670,6 +679,52 @@ class BackupRestore {
             return EXIT_CODES.CANNOT_RESTORE_BACKUP;
         }
 
+        const controllerDir = tools.getControllerDir();
+
+        // check that the same controller version is installed as it is contained in backup
+        if (!force) {
+            const backupHostname = restore.config.system.hostname || hostname;
+
+            try {
+                const ioPackJson = fs.readJsonSync(path.join(controllerDir, 'io-package.json'));
+                const backupJson = JSON.parse(data);
+                const hostObj = backupJson.objects.find(obj => obj.id === `system.host.${backupHostname}`);
+                if (hostObj.value.common.installedVersion !== ioPackJson.common.version) {
+                    console.warn('The current version of js-controller differs from the version in the backup.');
+                    console.warn('The js-controller version of the backup can not be restored automatically.');
+                    console.warn(
+                        `To restore the js-controller version of the backup, execute npm i iobroker.js-controller@${hostObj.value.common.installedVersion} --production inside your ioBroker directory`
+                    );
+                    console.warn(
+                        'If you really want to restore the backup with the current installed js-controller, execute the restore command with the --force flag'
+                    );
+
+                    return EXIT_CODES.CANNOT_RESTORE_BACKUP;
+                }
+            } catch {
+                // ignore
+            }
+        }
+
+        const nodeModulePath = path.join(controllerDir, '..');
+        const nodeModuleDirs = fs.readdirSync(nodeModulePath);
```

Comment:
better use fileObjs = fs.readdirSync(__dirname, { withFileTypes: true }); ?! then you can really only check the directories and not get unintentially files too